### DISCUSSION TITLE
throttle repetitve notifications (power management applet)

### DISCUSSION THREAD

**Discussion by mc0e** [linuxmint/discussions#161] [Ideas]
When my bluetooth mouse gets below 25% of battery remaining, I start getting pop-up notifications via the power management applet, saying "Mouse battery low" and on a second line, "Wireless mouse is low in power".

The frequency of These notifications is variable, but at worst it makes the upper right part of the screen unusable, with notifications re-appearing again almost immediately after the last is removed.

I haven't looked into the code, but I think this is probably because the reported percentage of battery remaining is wandering back and forth across a threshold, tripping a reporting condition repeatedly.  I can see the power level using Using `bluetoothctl info | grep Battery`, and it's currently going back and forth mostly between 25% and 26%, but using the mouse more or less I can get it to go to 24% or 27%.  I think it's reporting when it drops to 25%.

Can I suggest:

1. Include the battery percentage in the notification text.
2. Limit notifications by time elapsed, and perhaps also by the size of the change in the battery level from the last report.
3. Make the threshold for reporting configurable.  25% is too high in this case.
4. if these notifications are suitably throttled, then they should grab attention more, and persist longer so they aren't missed, and they should be able to display when there is a full-screen application running.  If I'm watching a full-screen movie, I commonly don't see or hear notifications about my laptop battery till it powers down.

I've had problems with high frequency notifications of other sorts of events before, making the interface unusable, but right now I can't recall what that was.  There's a need for a general mechanism for throttling the speed of notifications of all sorts.

**Comment by AtomicRobotMan0101** [comment]
I had this thought as well - perhaps an "acknowledged" button (or pull-down) and the notice wont re-appear for 24 hours?

If so, perhaps another pull-down option of "suppress these notices".

I have a soundbar that connects via Bluetooth. Great for the doof-doof, but the pops ups are annoying.

**Reply by mc0e** [reply]
A longer pause after acknowledgement would be useful, but there's never a good reason for sending duplicates at sub-second intervals, acknowledged or not.   Even if the messages are changing, there needs to be a few seconds in between in order that controls in that corner area (like closing a window) are usable.

**Comment by anaximeno** [comment]
The wifi service also shows a lot of notifications, specially when trying to connect to a network (and not being able to successfully connect).
